Source/HttpMultipartParser/StreamingMultipartFormDataParser.cs

```diff
@@ -801,7 +807,7 @@ private async Task ParseFilePartAsync(Dictionary<string, string> parameters, Reb
                     // We also want to chop off the newline that is inserted by the protocl.
                     // We can do this by reducing endPos by the length of newline in this environment
                     // and encoding
-                    FileHandler(name, filename, contentType, contentDisposition, fullBuffer, endPos - bufferNewlineLength);
+                    FileHandler(name, filename, contentType, contentDisposition, fullBuffer, endPos - bufferNewlineLength, partNumber);

```

Comment:
Shouldn't the sequence number be incremented here as well?

```suggestion
                    FileHandler(name, filename, contentType, contentDisposition, fullBuffer, endPos - bufferNewlineLength, partNumber++);

```

Reply:
In this place, the end of the file is processed and partNumber changing doesn't matter.